src/StoreApi/Schemas/AbstractSchema.php

```diff
@@ -232,45 +232,13 @@ protected function get_store_currency_properties() {
 	}
 
 	/**
-	 * Prepares a list of store currency data to return in responses.
+	 * Adds currency data to an array of monetary values.
 	 *
-	 * @todo Core could use a more defined currency object format, making use of
-	 * constants for currency format rather than strings, and holding this type
-	 * of information instead of plugins/blocks needed to normalize things
-	 * themselves.
-	 *
-	 * @return array
+	 * @param array $values Monetary amounts.
+	 * @return array Monetary amounts with currency data appended.
 	 */
-	protected function get_store_currency_response() {
-		$position = get_option( 'woocommerce_currency_pos' );
-		$symbol   = html_entity_decode( get_woocommerce_currency_symbol() );
-		$prefix   = '';
-		$suffix   = '';
-
-		switch ( $position ) {
-			case 'left_space':
-				$prefix = $symbol . ' ';
-				break;
-			case 'left':
-				$prefix = $symbol;
-				break;
-			case 'right_space':
-				$suffix = ' ' . $symbol;
-				break;
-			case 'right':
-				$suffix = $symbol;
-				break;
-		}
-
-		return [
-			'currency_code'               => get_woocommerce_currency(),
-			'currency_symbol'             => $symbol,
-			'currency_minor_unit'         => wc_get_price_decimals(),
-			'currency_decimal_separator'  => wc_get_price_decimal_separator(),
-			'currency_thousand_separator' => wc_get_price_thousand_separator(),
-			'currency_prefix'             => $prefix,
-			'currency_suffix'             => $suffix,
-		];
+	protected function prepare_currency_response( $values ) {
+		return $this->extend->formatters->currency->format( $values );
```

Comment:
For now this is sufficient. But it'd be nice to have AbstractSchema explicitly have `Formatters` as an injected dependency. This is a very loosely held opinion.

Reply:
Wouldn't we also need to expose it to plugins?

Reply:
I'm not suggesting _instead of_ being a dependency in ExtendRestAPI. I'm just saying it'd also be a dependency on AbstractSchema. But again, loosely held opinion and is fine as is.